File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int profit = 0;
        int buy = INT_MAX;
        for(int i=0; i<n; i++){
            if(prices[i] < buy){
                buy = prices[i];
            }
            else{   
                int currProfit = prices[i] - buy;
                profit = max(currProfit, profit);
            }
        }
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        profit = 0
        least = int(1e4)
        for i in prices:
            if i < least:
                least = i
                max_profit = max(profit, max_profit)
                profit = 0
            if least < i:
                profit = max(profit, i-least) 
        else:
            max_profit = max(profit, max_profit)
        return max_profit

            